src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/DataFlowOperationVisitor.cs

```diff
@@ -1260,7 +1260,8 @@ private void PerformPredicateAnalysis(IOperation operation)
                 operation.Kind == OperationKind.Invocation ||
                 operation.Kind == OperationKind.Argument ||
                 operation.Kind == OperationKind.FlowCaptureReference ||
-                operation.Kind == OperationKind.IsPattern);
+                operation.Kind == OperationKind.IsPattern ||
+                operation.Kind == OperationKind.ReDimClause);
```

Comment:
This should not be added here. ReDim clause can never be a predicate returning boolean value.